tests/bluetooth/tester/src/audio/btp_cap.c

```diff
@@ -543,30 +542,26 @@ static int cap_broadcast_source_adv_setup(struct btp_bap_broadcast_local_source
 	struct bt_data base_ad[2];
 	struct bt_data per_ad;
 
-	err = bt_rand(&broadcast_id, BT_AUDIO_BROADCAST_ID_SIZE);
-	if (err) {
-		printk("Unable to generate broadcast ID: %d\n", err);
-
-		return -EINVAL;
-	}
-
-	*gap_settings = BIT(BTP_GAP_SETTINGS_DISCOVERABLE) |
-			BIT(BTP_GAP_SETTINGS_EXTENDED_ADVERTISING);
-	/* Setup extended advertising data */
-	net_buf_simple_add_le16(&ad_buf, BT_UUID_BROADCAST_AUDIO_VAL);
-	net_buf_simple_add_le24(&ad_buf, source->broadcast_id);
-	base_ad[0].type = BT_DATA_SVC_DATA16;
-	base_ad[0].data_len = ad_buf.len;
-	base_ad[0].data = ad_buf.data;
-	base_ad[1].type = BT_DATA_NAME_COMPLETE;
-	base_ad[1].data_len = sizeof(CONFIG_BT_DEVICE_NAME) - 1;
-	base_ad[1].data = CONFIG_BT_DEVICE_NAME;
-	err = tester_gap_create_adv_instance(&param, BTP_GAP_ADDR_TYPE_IDENTITY, base_ad, 2, NULL,
-					     0, gap_settings);
-	if (err != 0) {
-		LOG_DBG("Failed to create extended advertising instance: %d", err);
-
-		return -EINVAL;
+	if (tester_gap_ext_adv_get() == NULL) {
```

Comment:
Please also move the `struct bt_le_adv_param param = *BT_LE_EXT_ADV_NCONN;`, `NET_BUF_SIMPLE_DEFINE(ad_buf, BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE);` and `struct bt_data base_ad[2];` into the `if` clause, as they are only used within this. 

Please also a comment stating why we are checking `if (tester_gap_ext_adv_get() == NULL) {` as that is non-obvious

Reply:
Changed